src/openssl/engine.c

```diff
@@ -29,6 +29,7 @@
 
 #include "keys.h"
 #include "../keychain.h"
+#include "uv.h"
```

Comment:
engine shouldn't need any of the libuv stuff.
I think you just need to replace `uv_os_fd_t` with `tlsuv_sock_t` in `set_io_fd()`

Reply:
`UV_EINVAL` is from `uv.h` AFAICT. (it's refernced on line 171>

There are other errors without uv.h for S_IFREG and S_IFDIR which also disappear with uv.h ...